infra/ibmcloud/terraform/k8s-s390x-infra-setup/modules/vpc/vpc.tf

```diff
@@ -0,0 +1,343 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+resource "ibm_is_vpc" "vpc" {
+  name           = "k8s-s390x-vpc"
+  resource_group = var.resource_group_id
+}
+
+# VPC
+resource "ibm_is_public_gateway" "public_gw" {
+  name           = "k8s-s390x-public-gw"
+  vpc            = ibm_is_vpc.vpc.id
+  zone           = var.zone
+  resource_group = var.resource_group_id
+}
+
+# Subnet
+resource "ibm_is_subnet" "subnet" {
+  name                     = "k8s-s390x-subnet"
```

Comment:
most likely you can create a local variable of 
```hcl
locals {
  name = "k8s-s390x"
}
```

and for all these resource names, just simple update to use
```hcl
  name = local.name
```

Don't add the `-gw` or `-sg` or `-vpc`, etc.